src/hooks/data/preview/useAssetToYbToken.ts

```diff
@@ -0,0 +1,87 @@
+import { Address } from "wagmi"
+
+import { VaultType } from "@/lib/types"
+import { useBalancerAssetToYbToken } from "@/hooks/data/preview/useBalancerAssetToYbToken"
+import { useCurveAssetToYbToken } from "@/hooks/data/preview/useCurveAssetToYbToken"
+import { useTokenAssetToYbToken } from "@/hooks/data/preview/useTokenAssetToYbToken"
+import {
+  useIsCurveCompounder,
+  useIsTokenCompounder,
+} from "@/hooks/useVaultTypes"
+
+import { useTxSettings } from "@/store/txSettings"
+
+import { DEFAULT_SLIPPAGE } from "@/constant/env"
+
+export function useAssetToYbToken({
```

Comment:
I think the naming is not quite right on these hooks. I'd keep it simple, like `usePreviewDeposit` and `usePreviewRedeem`. What do you think?

Reply:
These methods can be used also to estimate the swap amount between the ybtoken to the asset and vice versa, that's why I named them like this. But for this specific use, I agree it's simpler to call them this way.